v1/pro1.json

```
@@ -0,0 +1,110 @@
+{
+  "name": "F(x)tec Pro1",
+  "codename": "pro1",
+  "unlock": ["unlock"],
+  "user_actions": {
+    "unlock": {
+      "title": "OEM unlock",
+      "description": "If you haven't done so already, make sure to OEM unlock your device first.",
+      "link": "https://source.android.com/devices/bootloader/locking_unlocking#unlocking-bootloader"
+    },
+    "bootloader": {
+      "title": "Reboot to Bootloader",
+      "description": "With the device powered off, press and hold the volume down and power buttons until the screen turns on. It should show a menu selection screen.",
+      "image": "phone_power_down",
+      "button": true
+    },
+    "recovery": {
+      "title": "Reboot to Recovery",
+      "description": "With the device still at the 'Fastboot Mode' screen (if not, power off the device and press and hold volume down and power buttons until the screen turns on), select the 'Recovery Mode' option by using the volume buttons and press the power button to enter.",
+      "image": "phone_power_down",
+      "button": true
+    }
+  },
+  "operating_systems": [
+    {
+      "name": "Ubuntu Touch",
+      "sanity_check": "Are you sure?",
+      "options": [
+        {
+          "var": "channel",
+          "name": "Channel",
+          "tooltip": "The release channel",
+          "link": "https://docs.ubports.com/en/latest/about/process/release-schedule.html",
+          "type": "select",
+          "remote_values": { "type": "systemimagechannels" }
+        },
+        {
+          "var": "wipe",
+          "name": "Wipe Userdata",
+          "tooltip": "Wipe personal data",
+          "type": "checkbox"
+        },
+        {
+          "var": "bootstrap",
+          "name": "Bootstrap",
+          "tooltip": "Flash system partitions using fastboot",
+          "type": "checkbox",
+          "value": true
+        }
+      ],
+      "prerequisites": [],
+      "steps": [
+        {
+          "type": "download",
+          "condition": {"var": "bootstrap", "value": true},
+          "group": "firmware",
+          "files": [
+            {
+              "url": "https://gitlab.com/ubports/community-ports/android9/fxtec-pro1/fxtec-pro1/-/jobs/865239242/artifacts/raw/out/boot.img",
+              "checksum": "2204c2478854d4dfd1bb16fae2651f80ae07f4af4452eb4c0ff9bbc6ee3f6414"
+            }
+          ]
```

```suggestion
            {
              "url": "https://gitlab.com/ubports/community-ports/android9/fxtec-pro1/fxtec-pro1/-/jobs/865239242/artifacts/raw/out/boot.img",
              "checksum": {
                "sum": "2204c2478854d4dfd1bb16fae2651f80ae07f4af4452eb4c0ff9bbc6ee3f6414",
                "algorithm": "sha256"
              }
            }
          ]
```

Are you sure you need to flash boot? It's usually only required on devices where something is *janky*. Also, is it ok not to flash recovery? Can the stock recovery handle our commandfile? Any other OSs using different recoveries?

---

This device uses A/B which doesn't have a recovery partition. The recovery is inside the boot image.

---

Ah, ok. You'll still have to commit my suggestion to fix the CI warning and then rebase on master to resolve the conflict. Else than that, i'm happy to merge this. Thank you!